javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -1283,9 +1254,162 @@ public String stringPrefix() {
     public String toString() {
         return mkString(stringPrefix() + "(", ", ", ")");
     }
+
+    static final class VectorTree<T> implements Serializable {
```

Comment:
It is possible to rename this class to BMT and move outside of `Vector` with package visibility? Just thought